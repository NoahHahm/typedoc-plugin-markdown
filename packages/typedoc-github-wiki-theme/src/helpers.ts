export function parseUrl(url: string) {
  return encodeURI('../wiki/' + url.replace('.md', ''));
}
