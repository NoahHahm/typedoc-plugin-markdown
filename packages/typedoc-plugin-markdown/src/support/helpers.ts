import {
  ContainerReflection,
  DeclarationReflection,
  ProjectReflection,
  ReflectionKind,
  SignatureReflection,
} from 'typedoc';
import { backTicks } from './els';

/**
 * Determines if current signature is a constructor
 */
export const isConstructor = (
  signature: DeclarationReflection | SignatureReflection,
) => {
  return signature.parent?.kindOf(ReflectionKind.Constructor);
};

export function getIndexHeadingLevel(
  reflection: DeclarationReflection | ProjectReflection,
) {
  if (
    reflection.kindOf([
      ReflectionKind.Project,
      ReflectionKind.Module,
      ReflectionKind.Namespace,
    ]) ||
    reflection.hasOwnDocument
  ) {
    return 2;
  }
  return 4;
}

export function getGroupHeadingLevel(container: ContainerReflection) {
  return container.hasOwnDocument ? 2 : 4;
}

export function getReflectionHeadingLevel(
  reflection: DeclarationReflection | SignatureReflection,
) {
  if (reflection.hasOwnDocument) {
    return 1;
  }
  if (reflection.kindOf(ReflectionKind.Constructor)) {
    return reflection.parent?.hasOwnDocument ? 2 : 4;
  }
  return reflection.parent?.hasOwnDocument ? 3 : 5;
}

export function getMemberHeadingLevel(
  reflection: DeclarationReflection | SignatureReflection,
) {
  return getReflectionHeadingLevel(reflection) + 1;
}

export function getReflectionTitle(reflection: DeclarationReflection) {
  const md: string[] = [];

  md.push(reflection.name);

  if (reflection.typeParameters) {
    const typeParameters = reflection.typeParameters
      .map((typeParameter) => typeParameter.name)
      .join(', ');
    md.push(`\\<${typeParameters}\\>`);
  }

  return md.join('');
}

export function getFlags(reflection: DeclarationReflection) {
  if (reflection.flags?.length > 0 && !reflection.flags.isRest) {
    return reflection.flags.map((flag) => backTicks(flag)).join(' ');
  }
  return null;
}
