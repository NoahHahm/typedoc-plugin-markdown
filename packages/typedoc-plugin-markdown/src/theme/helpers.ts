/**
 *
 * A set of helpers that parses TypeDoc models to be consumed by theme resources.
 *
 * @module
 */

import {
  DeclarationReflection,
  ProjectReflection,
  ReflectionKind,
} from 'typedoc';
import { backTicks } from '../support/elements';
import { escapeChars } from '../support/utils';

export function getDeclarationType(declaration: DeclarationReflection) {
  if (declaration.getSignature) {
    return declaration.getSignature.type;
  }
  if (declaration.setSignature) {
    return declaration.setSignature.type;
  }
  return declaration.type;
}

export function getProjectDisplayName(
  project: ProjectReflection,
  includeVersion: boolean,
): string {
  const version =
    includeVersion && project.packageVersion
      ? ` - v${project.packageVersion}`
      : '';
  return `${project.name}${version ? version : ''}`;
}

export function hasIndex(
  reflection: DeclarationReflection | ProjectReflection,
) {
  return reflection.groups?.some((group) => group.allChildrenHaveOwnDocument());
}

export function hasTOC(reflection: DeclarationReflection | ProjectReflection) {
  return (
    reflection.kindOf([
      ReflectionKind.Project,
      ReflectionKind.Module,
      ReflectionKind.Namespace,
    ]) &&
    reflection.groups?.some((group) => !group.allChildrenHaveOwnDocument())
  );
}

export function isGroupKind(reflection: DeclarationReflection) {
  return reflection.kindOf([
    ReflectionKind.Class,
    ReflectionKind.Interface,
    ReflectionKind.Enum,
    ReflectionKind.Function,
    ReflectionKind.Variable,
    ReflectionKind.TypeAlias,
  ]);
}

export function getModifier(reflection: DeclarationReflection) {
  if (reflection.flags.isAbstract) {
    return 'abstract';
  }
  if (reflection.flags.isPrivate) {
    return 'private';
  }
  if (reflection.flags.isReadonly) {
    return 'readonly';
  }
  if (reflection.flags.isStatic) {
    return 'static';
  }
  if (reflection.getSignature) {
    return 'get';
  }
  if (reflection.setSignature) {
    return 'set';
  }
  return null;
}

export const KEYWORD_MAP = {
  [ReflectionKind.Class]: 'class',
  [ReflectionKind.Interface]: 'interface',
  [ReflectionKind.Enum]: 'enum',
  [ReflectionKind.TypeAlias]: 'type',
};

export function getMemberTitle(reflection: DeclarationReflection) {
  const md: string[] = [];

  const name: string[] = [];

  const flagsToInclude = ['Abstract'];

  const flags = reflection.flags.filter((flag) =>
    flagsToInclude.includes(flag),
  );

  if (flags.length) {
    name.push(flags.map((flag) => backTicks(flag.toLowerCase())).join(' '));
    name.push(' ');
  }

  name.push(`${escapeChars(reflection.name)}`);

  if (reflection.typeParameters) {
    const typeParameters = reflection.typeParameters
      .map((typeParameter) => typeParameter.name)
      .join(', ');
    name.push(`${backTicks(`<${typeParameters}>`)}`);
  }

  md.push(name.join(''));

  return md.join(': ');
}
