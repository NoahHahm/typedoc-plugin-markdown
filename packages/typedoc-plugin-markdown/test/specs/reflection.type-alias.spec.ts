import {
  FixtureOutputDir,
  FixtureOutputFileStrategy,
} from '../__utils__/config';
import { expectFileToEqual } from '../__utils__/helpers';

describe(`Type Alias Reflection`, () => {
  test(`should compile primitive type`, () => {
    expectFileToEqual(
      FixtureOutputDir.Reflections,
      FixtureOutputFileStrategy.Members,
      'type-aliases/PrimitiveType.md',
    );
  });

  test(`should compile array type`, () => {
    expectFileToEqual(
      FixtureOutputDir.Reflections,
      FixtureOutputFileStrategy.Members,
      'type-aliases/ArrayType.md',
    );
  });

  test(`should compile union type`, () => {
    expectFileToEqual(
      FixtureOutputDir.Reflections,
      FixtureOutputFileStrategy.Members,
      'type-aliases/UnionType.md',
    );
  });

  test(`should compile string literal type`, () => {
    expectFileToEqual(
      FixtureOutputDir.Reflections,
      FixtureOutputFileStrategy.Members,
      'type-aliases/StringLiteralType.md',
    );
  });

  test(`should compile string tuple type`, () => {
    expectFileToEqual(
      FixtureOutputDir.Reflections,
      FixtureOutputFileStrategy.Members,
      'type-aliases/TupleType.md',
    );
  });

  test(`should compile intersection type`, () => {
    expectFileToEqual(
      FixtureOutputDir.Reflections,
      FixtureOutputFileStrategy.Members,
      'type-aliases/IntersectionType.md',
    );
  });

  test(`should compile conditional type`, () => {
    expectFileToEqual(
      FixtureOutputDir.Reflections,
      FixtureOutputFileStrategy.Members,
      'type-aliases/ConditionalType.md',
    );
  });

  test(`should compile type parameter type`, () => {
    expectFileToEqual(
      FixtureOutputDir.Reflections,
      FixtureOutputFileStrategy.Members,
      'type-aliases/TypeWithTypeParams.md',
    );
  });

  test(`should compile external reference type`, () => {
    expectFileToEqual(
      FixtureOutputDir.Reflections,
      FixtureOutputFileStrategy.Members,
      'type-aliases/ExternalReferenceType.md',
    );
  });

  test(`should compile partial mapped type`, () => {
    expectFileToEqual(
      FixtureOutputDir.Reflections,
      FixtureOutputFileStrategy.Members,
      'type-aliases/PartialMappedType.md',
    );
  });

  test(`should compile readonly mapped type`, () => {
    expectFileToEqual(
      FixtureOutputDir.Reflections,
      FixtureOutputFileStrategy.Members,
      'type-aliases/ReadonlyMappedType.md',
    );
  });

  test(`should compile function type`, () => {
    expectFileToEqual(
      FixtureOutputDir.Reflections,
      FixtureOutputFileStrategy.Members,
      'type-aliases/FunctionType.md',
    );
  });

  test(`should compile external link symbol type`, () => {
    expectFileToEqual(
      FixtureOutputDir.Reflections,
      FixtureOutputFileStrategy.Members,
      'type-aliases/TypeWithExternalSymbolLinkMapping.md',
    );
  });

  test(`should compile external literal type`, () => {
    expectFileToEqual(
      FixtureOutputDir.Reflections,
      FixtureOutputFileStrategy.Members,
      'type-aliases/LiteralType.md',
    );
  });
});
