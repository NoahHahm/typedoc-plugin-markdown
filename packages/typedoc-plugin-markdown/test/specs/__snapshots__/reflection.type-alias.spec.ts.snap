// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type Alias Reflection should compile array type: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** ∙ [Readme](../README.md) ∙ [API](../exports.md)

***

[API](../exports.md) > ArrayType

# Type alias: ArrayType

> **ArrayType**: \`string\`[]

Comments for ArrayType

## Source

[types.ts:12](http://source-url)
"
`;

exports[`Type Alias Reflection should compile array type: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs**

***

[API](../index.md) > ArrayType

# ArrayType

\`\`\`ts
type ArrayType: string[];
\`\`\`

Comments for ArrayType

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Type Alias Reflection should compile conditional type: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** ∙ [Readme](../README.md) ∙ [API](../exports.md)

***

[API](../exports.md) > ConditionalType

# Type alias: ConditionalType\`<T>\`

> **ConditionalType**\\<\`T\`\\>: \`T\` extends \`string\` ? \`"string"\` : \`T\` extends \`number\` ? \`"number"\` : \`T\` extends \`boolean\` ? \`"boolean"\` : \`T\` extends \`undefined\` ? \`"undefined"\` : \`"object"\`

Comments for ConditionalType

## Type parameters

| Parameter |
| :------ |
| \`T\` |

## Source

[types.ts:46](http://source-url)
"
`;

exports[`Type Alias Reflection should compile conditional type: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs**

***

[API](../index.md) > ConditionalType

# ConditionalType\`<T>\`

\`\`\`ts
type ConditionalType<T>: T extends string ? "string" : T extends number ? "number" : T extends boolean ? "boolean" : T extends undefined ? "undefined" : "object";
\`\`\`

Comments for ConditionalType

## Type parameters

| Parameter |
| :------ |
| \`T\` |

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Type Alias Reflection should compile external link symbol type: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** ∙ [Readme](../README.md) ∙ [API](../exports.md)

***

[API](../exports.md) > TypeWithExternalSymbolLinkMapping

# Type alias: TypeWithExternalSymbolLinkMapping

> **TypeWithExternalSymbolLinkMapping**: [\`Application\`]( https://typedoc.org/api/classes/Application.html )

Comments for TypeWithExternalSymbolLinkMapping

## Source

[types.ts:88](http://source-url)
"
`;

exports[`Type Alias Reflection should compile external link symbol type: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs**

***

[API](../index.md) > TypeWithExternalSymbolLinkMapping

# TypeWithExternalSymbolLinkMapping

\`\`\`ts
type TypeWithExternalSymbolLinkMapping: Application;
\`\`\`

Comments for TypeWithExternalSymbolLinkMapping

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Type Alias Reflection should compile external literal type: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** ∙ [Readme](../README.md) ∙ [API](../exports.md)

***

[API](../exports.md) > LiteralType

# Type alias: LiteralType

> **LiteralType**: \`object\`

Comments for LiteralType

## Type declaration

### x

> **x**: \`string\`

comment for x

### y

> **y**: \`object\`

comment for y

### y.x

> **y.x**: \`string\`

comment for y.x

### y.y

> **y.y**: \`boolean\` \\| \`string\`

comment for y.y

### y.z

> **y.z**: (\`x\`) => \`string\`

comment for y.z

#### Parameters

▪ **x**: \`string\`

### z

> **z**: (\`x\`) => \`string\`

#### Parameters

▪ **x**: \`string\`

### accessorA

> **\`get\`** **accessorA**(): \`Promise\`\\<\`string\`\\>

Comments for accessorA getter

> **\`set\`** **accessorA**(\`x\`): \`void\`

Comments for accessorA setter

#### Parameters

▪ **x**: \`Promise\`\\<\`string\`\\>

### accessorB

> **\`get\`** **accessorB**(): \`string\`

> **\`set\`** **accessorB**(\`x\`): \`void\`

#### Parameters

▪ **x**: \`string\`

### someFunction()

Comments for someFunction

#### Parameters

▪ **param**: \`string\`

## Source

[types.ts:93](http://source-url)
"
`;

exports[`Type Alias Reflection should compile external literal type: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs**

***

[API](../index.md) > LiteralType

# LiteralType

\`\`\`ts
type LiteralType: {
  x: string;
  y: {
     x: string;
     y: boolean | string;
     z: (x) => string;
  };
  z: (x) => string;
  get set accessorA: Promise<string>;
  get set accessorB: string;
  someFunction: Promise<any>;
};
\`\`\`

Comments for LiteralType

## Type declaration

| Member | Type | Description |
| :------ | :------ | :------ |
| \`x\` | \`string\` | comment for x |
| \`y\` | \\{   \`x\`: \`string\`;   \`y\`: \`boolean\` \\| \`string\`;   \`z\`: (\`x\`) => \`string\`;   } | comment for y |
| \`y.x\` | \`string\` | comment for y.x |
| \`y.y\` | \`boolean\` \\| \`string\` | comment for y.y |
| \`y.z\` | (\`x\`) => \`string\` | - |
| \`z\` | (\`x\`) => \`string\` | - |
| \`get accessorA\` | \`Promise\`\\<\`string\`\\> | Comments for accessorA getter |
| \`set accessorA\` | \`void\` | Comments for accessorA setter |
| \`get accessorB\` | \`string\` | - |
| \`set accessorB\` | \`void\` | - |
| \`someFunction\` | \`Promise\`\\<\`any\`\\> | Comments for someFunction |

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Type Alias Reflection should compile external reference type: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** ∙ [Readme](../README.md) ∙ [API](../exports.md)

***

[API](../exports.md) > ExternalReferenceType

# Type alias: ExternalReferenceType

> **ExternalReferenceType**: [\`ClassWithTypeParameters\`](../classes/ClassWithTypeParameters.md)\\<\`"x"\`, \`"y"\`\\>

Comments for ExternalReferenceType

## Source

[types.ts:64](http://source-url)
"
`;

exports[`Type Alias Reflection should compile external reference type: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs**

***

[API](../index.md) > ExternalReferenceType

# ExternalReferenceType

\`\`\`ts
type ExternalReferenceType: ClassWithTypeParameters<"x", "y">;
\`\`\`

Comments for ExternalReferenceType

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Type Alias Reflection should compile function type: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** ∙ [Readme](../README.md) ∙ [API](../exports.md)

***

[API](../exports.md) > FunctionType

# Type alias: FunctionType

> **FunctionType**: (\`name\`, \`value\`) => \`void\`

Comments for FunctionType

## Parameters

▪ **name**: \`string\`

▪ **value**: \`unknown\`

## Source

[types.ts:83](http://source-url)
"
`;

exports[`Type Alias Reflection should compile function type: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs**

***

[API](../index.md) > FunctionType

# FunctionType

\`\`\`ts
type FunctionType: (name, value) => void;
\`\`\`

Comments for FunctionType

## Parameters

| Parameter | Type |
| :------ | :------ |
| \`name\` | \`string\` |
| \`value\` | \`unknown\` |

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Type Alias Reflection should compile index access type: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** ∙ [Readme](../README.md) ∙ [API](../exports.md)

***

[API](../exports.md) > IndexAccessType

# Type alias: IndexAccessType

> **IndexAccessType**: [\`ArrayOfStuff\`](ArrayOfStuff.md)\\[\`number\`\\]

Comments for IndexAccessType

## Source

[types.ts:140](http://source-url)
"
`;

exports[`Type Alias Reflection should compile index access type: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs**

***

[API](../index.md) > IndexAccessType

# IndexAccessType

\`\`\`ts
type IndexAccessType: ArrayOfStuff\\[number\\];
\`\`\`

Comments for IndexAccessType

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Type Alias Reflection should compile intersection type: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** ∙ [Readme](../README.md) ∙ [API](../exports.md)

***

[API](../exports.md) > IntersectionType

# Type alias: IntersectionType

> **IntersectionType**: [\`TupleType\`](TupleType.md) & [\`ArrayType\`](ArrayType.md) & \`object\`

Comments for IntersectionType

## Type declaration

### bar

> **bar**: \`number\`

## Source

[types.ts:38](http://source-url)
"
`;

exports[`Type Alias Reflection should compile intersection type: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs**

***

[API](../index.md) > IntersectionType

# IntersectionType

\`\`\`ts
type IntersectionType: TupleType & ArrayType & {
  bar: number;
};
\`\`\`

Comments for IntersectionType

## Type declaration

| Member | Type | Description |
| :------ | :------ | :------ |
| \`bar\` | \`number\` | - |

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Type Alias Reflection should compile partial mapped type: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** ∙ [Readme](../README.md) ∙ [API](../exports.md)

***

[API](../exports.md) > PartialMappedType

# Type alias: PartialMappedType\`<T>\`

> **PartialMappedType**\\<\`T\`\\>: \`{ [P in keyof T]?: T[P] }\`

Comments for PartialMappedType

## Type parameters

| Parameter |
| :------ |
| \`T\` |

## Source

[types.ts:69](http://source-url)
"
`;

exports[`Type Alias Reflection should compile partial mapped type: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs**

***

[API](../index.md) > PartialMappedType

# PartialMappedType\`<T>\`

\`\`\`ts
type PartialMappedType<T>: { [P in keyof T]?: T[P] };
\`\`\`

Comments for PartialMappedType

## Type parameters

| Parameter |
| :------ |
| \`T\` |

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Type Alias Reflection should compile primitive type: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** ∙ [Readme](../README.md) ∙ [API](../exports.md)

***

[API](../exports.md) > PrimitiveType

# Type alias: PrimitiveType

> **PrimitiveType**: \`boolean\`

Comments for PrimitiveType

## Source

[types.ts:7](http://source-url)
"
`;

exports[`Type Alias Reflection should compile primitive type: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs**

***

[API](../index.md) > PrimitiveType

# PrimitiveType

\`\`\`ts
type PrimitiveType: boolean;
\`\`\`

Comments for PrimitiveType

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Type Alias Reflection should compile readonly mapped type: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** ∙ [Readme](../README.md) ∙ [API](../exports.md)

***

[API](../exports.md) > ReadonlyMappedType

# Type alias: ReadonlyMappedType\`<T>\`

> **ReadonlyMappedType**\\<\`T\`\\>: \`{ readonly [P in keyof T]: T[P] }\`

Comments for ReadonlyMapedType

## Type parameters

| Parameter |
| :------ |
| \`T\` |

## Source

[types.ts:76](http://source-url)
"
`;

exports[`Type Alias Reflection should compile readonly mapped type: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs**

***

[API](../index.md) > ReadonlyMappedType

# ReadonlyMappedType\`<T>\`

\`\`\`ts
type ReadonlyMappedType<T>: { readonly [P in keyof T]: T[P] };
\`\`\`

Comments for ReadonlyMapedType

## Type parameters

| Parameter |
| :------ |
| \`T\` |

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Type Alias Reflection should compile string literal type: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** ∙ [Readme](../README.md) ∙ [API](../exports.md)

***

[API](../exports.md) > StringLiteralType

# Type alias: StringLiteralType

> **StringLiteralType**: \`" "\` \\| \`"string"\` \\| \`"string|with|pipes"\` \\| "string\\\`with\\\`backticks" \\| \`"<foo>"\` \\| \`"*"\`

Comments for StringLiteralType

## Source

[types.ts:22](http://source-url)
"
`;

exports[`Type Alias Reflection should compile string literal type: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs**

***

[API](../index.md) > StringLiteralType

# StringLiteralType

\`\`\`ts
type StringLiteralType: 
  | " "
  | "string"
  | "string|with|pipes"
  | "string\\with\\backticks"
  | "<foo>"
  | "*";
\`\`\`

Comments for StringLiteralType

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Type Alias Reflection should compile string tuple type: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** ∙ [Readme](../README.md) ∙ [API](../exports.md)

***

[API](../exports.md) > TupleType

# Type alias: TupleType

> **TupleType**: [\`string\`, \`number\`]

Comments for TupleType

## Source

[types.ts:33](http://source-url)
"
`;

exports[`Type Alias Reflection should compile string tuple type: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs**

***

[API](../index.md) > TupleType

# TupleType

\`\`\`ts
type TupleType: [string, number];
\`\`\`

Comments for TupleType

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Type Alias Reflection should compile type parameter type: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** ∙ [Readme](../README.md) ∙ [API](../exports.md)

***

[API](../exports.md) > TypeWithTypeParams

# Type alias: TypeWithTypeParams\`<T, R>\`

> **TypeWithTypeParams**\\<\`T\`, \`R\`\\>: [\`T\`, \`R\`]

Comments for TypeWithTypeParams

## Type parameters

| Parameter |
| :------ |
| \`T\` |
| \`R\` |

## Source

[types.ts:59](http://source-url)
"
`;

exports[`Type Alias Reflection should compile type parameter type: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs**

***

[API](../index.md) > TypeWithTypeParams

# TypeWithTypeParams\`<T, R>\`

\`\`\`ts
type TypeWithTypeParams<T, R>: [T, R];
\`\`\`

Comments for TypeWithTypeParams

## Type parameters

| Parameter |
| :------ |
| \`T\` |
| \`R\` |

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Type Alias Reflection should compile union type: (Output File Strategy "members") (Option Group "1") 1`] = `
"**typedoc-stubs** ∙ [Readme](../README.md) ∙ [API](../exports.md)

***

[API](../exports.md) > UnionType

# Type alias: UnionType

> **UnionType**: \`string\` \\| \`boolean\` \\| \`object\`

Comments for UnionType

## Source

[types.ts:17](http://source-url)
"
`;

exports[`Type Alias Reflection should compile union type: (Output File Strategy "members") (Option Group "2") 1`] = `
"**typedoc-stubs**

***

[API](../index.md) > UnionType

# UnionType

\`\`\`ts
type UnionType: string | boolean | {
  z: string;
};
\`\`\`

Comments for UnionType

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;
